snuba/manual_jobs/extract_span_data.py

```diff
@@ -0,0 +1,121 @@
+from typing import Any, Mapping, Optional
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.manual_jobs import Job, JobLogger, JobSpec
+
+
+class ExtractSpanData(Job):
+    def __init__(self, job_spec: JobSpec) -> None:
+        self.__validate_job_params(job_spec.params)
+        super().__init__(job_spec)
+
+    def __validate_job_params(self, params: Optional[Mapping[Any, Any]]) -> None:
+        assert params
+        required_params = [
+            "organization_ids",
+            "start_timestamp",
+            "end_timestamp",
+            "table_name",
+            "limit",
+            "gcp_bucket_name",
+            "output_file_path",
+        ]
+        for param in required_params:
+            assert param in params
```

Comment:
I think you can use Counter() to compare two unordered lists.
```
Python 3.11.8 (main, Aug 16 2024, 10:27:21) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections import Counter
>>> l1 = ["one", "two", "three"]
>>> l2 = ["two", "one", "three"]
>>> assert Counter(l1) == Counter(l2), "Lists must be same!"
>>> l1.append("four")
>>> assert Counter(l1) == Counter(l2), "Lists must be same!"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: Lists must be same!
```
Also, consider adding a message along with `assert` to make debugging easier.